### PR TITLE
refactored LDAPPassword to reuse more code

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Follow-up from https://github.com/openshift/origin/pull/5145

@liggitt PTAL

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@liggitt this is rebased and ready

[test]

**Comment by stevekuznetsov** [comment]
flake [here](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/7552/console):
```
Get https://copr.fedoraproject.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo: dial tcp 209.132.184.54:443: no route to host
```
https://github.com/openshift/origin/issues/6040

**Comment by stevekuznetsov** [comment]
re[test]

File: pkg/auth/authenticator/password/ldappassword/ldap.go [hunk 1]
```diff
@@ -122,27 +109,20 @@ func (a *Authenticator) getIdentity(username, password string) (authapi.UserIden
 		nil,          // controls
 	)
 
-	glog.V(4).Infof("searching for %s", filter)
-	results, err := l.Search(searchRequest)
+	entry, err := ldaputil.QueryForUniqueEntry(a.options.ClientConfig, searchRequest)
```
Comment (liggitt) [inline pkg/auth/authenticator/password/ldappassword/ldap.go]: this results in two server connections, we only want one

Comment (liggitt) [inline pkg/auth/authenticator/password/ldappassword/ldap.go]: are we calling `QueryForUniqueEntry` in a loop in the sync job? we should be reusing the connection... TLS establishment is expensive

**Comment by liggitt** [comment]
I don't think the refactor is worth it

**Comment by stevekuznetsov** [comment]
We try to cache as much data as possible but I think it's possible we connect multiple times. Is there a way to tell from outside of Origin?

**Comment by liggitt** [comment]
just on inspection, querying for a single group creates and tears down a new connection each time

**Pull request closed without merging**